src/ServiceInsight/MessageFlow/MessageFlowViewModel.cs

```diff
@@ -118,7 +118,11 @@ public void ShowSagaWindow()
             {
                 selection.SelectedMessage = SelectedMessage.Message;
             }
-            eventAggregator.PublishOnUIThread(SwitchToSagaWindow.Instance);
+
+            if (selection.SelectedMessage?.Sagas?.Count == 1)
```

Comment:
Can this just rely on `HasSaga`? Do we grey out the tab or something or do we just refuse to display the tab?

Reply:
This added check means if you click on the 'saga' part of the message node in Flow Diagram, it won't take you to the Saga View (if we have more than one saga to show).

You can't rely on HasSaga as if you have more than one, that'd return true.